x-pack/packages/kbn-elastic-assistant/impl/assistant/prompt_editor/system_prompt/system_prompt_settings_management/index.tsx

```diff
@@ -0,0 +1,218 @@
+  conversationsSettingsBulkActions,
+  setConversationsSettingsBulkActions,
+  defaultConnector,
+  handleSave,
+  onCancelClick,
+  resetSettings,
+}: Props) => {
+  const { isFlyoutOpen: editFlyoutVisible, openFlyout, closeFlyout } = useFlyoutModalVisibility();
+  const {
+    isFlyoutOpen: deleteConfirmModalVisibility,
+    openFlyout: openConfirmModal,
+    closeFlyout: closeConfirmModal,
+  } = useFlyoutModalVisibility();
+  const [deletedPrompt, setDeletedPrompt] = useState<Prompt | null>();
+
+  const onCreate = useCallback(() => {
+    onSelectedSystemPromptChange({
+      id: '',
+      content: '',
+      name: '',
+      promptType: 'system',
+    });
+    openFlyout();
+  }, [onSelectedSystemPromptChange, openFlyout]);
+
+  const { onSystemPromptSelectionChange, onSystemPromptDeleted } = useSystemPromptEditor({
+    setUpdatedSystemPromptSettings,
+    onSelectedSystemPromptChange,
+  });
+
+  const onEditActionClicked = useCallback(
+    (prompt: Prompt) => {
+      onSystemPromptSelectionChange(prompt);
+      openFlyout();
+    },
+    [onSystemPromptSelectionChange, openFlyout]
+  );
+
+  const onDeleteActionClicked = useCallback(
+    (prompt: Prompt) => {
+      setDeletedPrompt(prompt);
+      onSystemPromptDeleted(prompt.id);
+      openConfirmModal();
+    },
+    [onSystemPromptDeleted, openConfirmModal]
+  );
+
+  const onDeleteCancelled = useCallback(() => {
+    setDeletedPrompt(null);
+    closeConfirmModal();
+    onCancelClick();
+  }, [closeConfirmModal, onCancelClick]);
+
+  const onDeleteConfirmed = useCallback(() => {
+    closeConfirmModal();
+    handleSave(true);
+    setConversationsSettingsBulkActions({});
+  }, [closeConfirmModal, handleSave, setConversationsSettingsBulkActions]);
+
+  const onSaveCancelled = useCallback(() => {
+    closeFlyout();
+    onCancelClick();
+  }, [closeFlyout, onCancelClick]);
+
+  const onSaveConfirmed = useCallback(() => {
+    closeFlyout();
+    handleSave(true);
+    setConversationsSettingsBulkActions({});
+  }, [closeFlyout, handleSave, setConversationsSettingsBulkActions]);
+
+  const confirmationTitle = useMemo(
+    () =>
+      deletedPrompt?.name
+        ? i18n.DELETE_SYSTEM_PROMPT_MODAL_TITLE(deletedPrompt?.name)
+        : i18n.DELETE_SYSTEM_PROMPT_MODAL_DEFAULT_TITLE,
+    [deletedPrompt?.name]
+  );
+
+  const { getColumns, getSystemPromptsList } = useSystemPromptTable();
+
+  const columns = useMemo(
+    () => getColumns({ onEditActionClicked, onDeleteActionClicked }),
+    [getColumns, onEditActionClicked, onDeleteActionClicked]
+  );
+  const systemPromptListItems = useMemo(
+    () =>
+      getSystemPromptsList({
+        connectors,
+        conversationSettings,
+        defaultConnector,
+        systemPromptSettings,
+      }),
+    [getSystemPromptsList, connectors, conversationSettings, defaultConnector, systemPromptSettings]
+  );
+
+  const pagination = useMemo(
+    () => ({
+      pageIndex: DEFAULT_PAGE_INDEX,
+      pageSize: DEFAULT_PAGE_SIZE,
```

Comment:
In quick prompts table and system prompts table, I checked with @bojanasan to give it a bigger default page size as currently new item is added to the bottom of the list and we do not have timestamp to sort on.
<img width="2558" alt="Screenshot 2024-06-27 at 20 59 30" src="https://github.com/elastic/kibana/assets/6295984/af671d8e-9ba9-402f-a322-a97531e2192b">